### ISSUE TITLE
🐛 Inconsistent Hint Display for Vue

### ISSUE THREAD

**Issue by hexadecimal233** [biomejs/biome-vscode#834]
### VS Code version

1.105.1

### Extension version

3.3.0

### Biome version

2.3.1

### Operating system

- [x] Windows
- [ ] macOS
- [ ] Linux

### Description

Biome is displaying hints at wrong lines inside vue files.
This is occuring in my [repo](https://github.com/hexadecimal233/cloudie)

### Steps to reproduce

1. Find a Vue file with some error/lintings
2. The problem happens

Screenshots:

The error:
<img width="1070" height="535" alt="Image" src="https://github.com/user-attachments/assets/e806414f-dcf4-4c4e-9e33-712546588d22" />

Display:
<img width="1851" height="397" alt="Image" src="https://github.com/user-attachments/assets/1a4488b8-3de1-4c84-bb49-acd5287582b9" />

### Expected behavior

The hints should be displayed where they should.

### Does this issue occur when using the CLI directly?

Not sure / Not applicable

### Link to a minimal reproduction

Any Vue file should trigger the problem.

### Logs

```plain text

```

**Comment by lucsorel** [comment]
I am also facing this issue since I updated `biome` version to `2.3.x`:
<img width="712" height="677" alt="Image" src="https://github.com/user-attachments/assets/43b50b24-c5c8-43dd-bbc1-4a8b939f3a3f" />

Before 2.3.x, hints were displayed at the proper places (with the known issues that biome would complain for unused functions, which were used in templates - I knew that and was fine with it). But know, it makes the tool less relevant :/

- codium version: 1.105.06808 (commit: 2e80ae9dbe80f2f38d6b895382def5a958b7421f, date: 2025-10-10T16:56:00.728Z, Electron: 37.6.0, Chromium: 138.0.7204.251, Node.js: 22.19.0, V8: 13.8.258.32-electron.0, OS: Linux x64 6.8.0-86-generic)

- Biome extension version: 3.3.0

- Biome version: 2.3.4

- operating system: Linux Ubuntu

**Comment by jonboiser** [comment]
For me, it seems like the hint is consistently placed 25 characters before the actual code. 

For example, the red squiggle is for `noUnusedVariables` on foo, but appears at the column 1 of the line (the opening `/*` of the comment). The actual violation is past the comment, on col 26-28

<img width="588" height="100" alt="Image" src="https://github.com/user-attachments/assets/c920f5c0-74d2-4159-99c5-6c38e8c53aac" />

**Comment by hexadecimal233** [comment]
For me this was fixed by updating biome and turning on HTML experiments in `biome.json`, and seems they are now working on this stuff.

```json
"html": {
    "formatter": { "enabled": true, "selfCloseVoidElements": "always" },
    "experimentalFullSupportEnabled": true
  }
```

**Comment by lucsorel** [comment]
> For me this was fixed by updating biome and turning on HTML experiments in biome.json,
>"html": {
>    "formatter": { "enabled": true, "selfCloseVoidElements": "always" },
>    "experimentalFullSupportEnabled": true
>  }

Hi @hexadecimal233 

I have updated biome to 2.3.7 and activated html experiments and it doesn't change anything on this issue, unfortunately:

<img width="986" height="332" alt="Image" src="https://github.com/user-attachments/assets/05e98334-34e8-4475-8e68-4a4d11d39600" />


> and seems they are now working on this stuff.

Do you mean that the development team of biome is aware of this issue and working on it?

**Comment by hexadecimal233** [comment]
> Do you mean that the development team of biome is aware of this issue and working on it?

This was fixed for most, so check the extension is using the correct version and config of biome ig

**Comment by lucsorel** [comment]
hello @hexadecimal233 

Unfortunately, I still experience the same misplaced squiggles using:
- the 3.3.0 version of the vsCode extension
- the 2.3.8 of the binary
- with or without the `"html": {...}` settings you mentioned https://github.com/biomejs/biome-vscode/issues/834#issuecomment-3567569948

<img width="888" height="578" alt="Image" src="https://github.com/user-attachments/assets/7a3e4712-bc5e-44a1-b3a7-bd1ec3be85c3" />

> This was fixed for most, so check the extension is using the correct version and config of biome ig

In the bottom left corner of my vscode, I can see that the biome extension is using the 2.3.8 version:

<img width="1006" height="268" alt="Image" src="https://github.com/user-attachments/assets/6a1aaf7f-328a-49e7-bfc9-db1144ae891f" />

Is this the proper way to check the version used?

Also, I did not understand what you meant by "config of biome ig" (the "ig" part).

Here is the `biome.jsonc` used by the project in which I experience the misplaced wiggles:

```js
{
  // https://biomejs.dev/reference/configuration/
  "$schema": "https://biomejs.dev/schemas/2.3.8/schema.json",
  // https://biomejs.dev/guides/integrate-in-vcs/
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "root": "..",
    "useIgnoreFile": true,
    "defaultBranch": "develop"
  },
  "formatter": {
    "enabled": true,
    "indentStyle": "space",
    "lineWidth": 100
  },
  "css": {
    "formatter": {
      "indentStyle": "space",
      "lineWidth": 100
    }
  },
  "javascript": {
    "formatter": {
      "indentStyle": "space",
      "quoteStyle": "single",
      "semicolons": "asNeeded",
      "trailingCommas": "es5"
    }
  },
  "linter": {
    "enabled": true,
    "includes": ["**/*.js", "**/*.ts"],
    "rules": {
      "recommended": true
    }
  },
  "json": {
    "formatter": {
      "enabled": true,
      "indentStyle": "space",
      "lineWidth": 100
    }
  },
  "assist": {
    "enabled": true
  },
  "files": {
    "maxSize": 10485760 // 10 Mo for some e2e fixture files (structure evaluation)
  }
}
```
